nbs/99_manuscript/coefs_comp/py/08_00-gtex_whole_blood-general_plots.py

```diff
@@ -0,0 +1,168 @@
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.5
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown] tags=[]
+# # Description
+
+# %% [markdown] tags=[]
+# It generates different general plots to compare coefficient values from Pearson, Spearman and Clustermatch, such as their distribution.
+#
+# In `Settings` below, the data set and other options (such as tissue for GTEx) are specified.
+
+# %% [markdown] tags=[]
+# # Modules
+
+# %% tags=[]
+import pandas as pd
+from scipy import stats
+import seaborn as sns
+
+from clustermatch.plots import plot_histogram, plot_cumulative_histogram, jointplot
+from clustermatch import conf
+
+# %% [markdown] tags=[]
+# # Settings
+
+# %% tags=[]
+DATASET_CONFIG = conf.GTEX
+GTEX_TISSUE = "whole_blood"
+GENE_SEL_STRATEGY = "var_pc_log2"
+
+# %% tags=[]
+# this is used for the cumulative histogram
+GENE_PAIRS_PERCENT = 0.70
+
+# %% [markdown] tags=[]
+# # Paths
+
+# %% tags=[]
+assert (
+    conf.MANUSCRIPT["BASE_DIR"] is not None and conf.MANUSCRIPT["BASE_DIR"].exists()
+), "Manuscript dir not set"
+
+# %% tags=[]
+OUTPUT_FIGURE_DIR = (
+    conf.MANUSCRIPT["FIGURES_DIR"] / "coefs_comp" / f"gtex_{GTEX_TISSUE}"
+)
+OUTPUT_FIGURE_DIR.mkdir(parents=True, exist_ok=True)
+display(OUTPUT_FIGURE_DIR)
+
+# %% tags=[]
+INPUT_CORR_FILE_TEMPLATE = (
+    DATASET_CONFIG["SIMILARITY_MATRICES_DIR"]
+    / DATASET_CONFIG["SIMILARITY_MATRIX_FILENAME_TEMPLATE"]
+)
+display(INPUT_CORR_FILE_TEMPLATE)
+
+# %% tags=[]
+INPUT_FILE = DATASET_CONFIG["SIMILARITY_MATRICES_DIR"] / str(
+    INPUT_CORR_FILE_TEMPLATE
+).format(
+    tissue=GTEX_TISSUE,
+    gene_sel_strategy=GENE_SEL_STRATEGY,
+    corr_method="all",
+)
+display(INPUT_FILE)
+
+assert INPUT_FILE.exists()
+
+# %% [markdown] tags=[]
+# # Data
+
+# %% tags=[]
+df = pd.read_pickle(INPUT_FILE)
+
+# %% tags=[]
+df.shape
+
+# %% tags=[]
+df.head()
+
+# %% [markdown] tags=[]
+# ## Data stats
+
+# %% tags=[]
+df.describe().applymap(str)
+
+# %% tags=[]
+# skewness
+df.apply(lambda x: stats.skew(x))
+
+# %% [markdown] tags=[]
+# # Histogram plot
+
+# %% tags=[]
+with sns.plotting_context("talk", font_scale=1.0):
+    plot_histogram(df, output_dir=OUTPUT_FIGURE_DIR, fill=False)
+
+# %% [markdown] tags=[]
+# Coefficients' values distribute very differently. Clustermatch is skewed to the left, whereas Pearson and specially Spearman seem more uniform.
+
+# %% [markdown] tags=[]
+# # Cumulative histogram plot
+
+# %% [markdown] tags=[]
+# I include also a cumulative histogram without specifying `bins`.
+
+# %% tags=[]
+with sns.plotting_context("talk", font_scale=1.0):
+    plot_cumulative_histogram(df, GENE_PAIRS_PERCENT, output_dir=OUTPUT_FIGURE_DIR)
+
+# %% [markdown] tags=[]
+# # Joint plots comparing each coefficient
+
+# %% tags=[]
+with sns.plotting_context("talk", font_scale=1.0):
```

Comment:
Actually, there are more gene pairs with higher Clustermatch and lower Pearson. This is more clearly viewed in the UpSet plot, where we take the top 30% of gene pairs from each coefficient. Here this hexbin plot might not be that clear, because the distribution of Clutermatch and Pearson is very different (Clustermatch is more skewed).

Reply:
This is the figure in the manuscript just in case:

![image](https://user-images.githubusercontent.com/172687/151190561-42ab9b6f-b612-4bea-b839-9b787c611f90.png)